core/backend/src/rpc-impl/IModelTileRpcImpl.ts

```diff
@@ -189,31 +192,43 @@ export class IModelTileRpcImpl extends RpcInterface implements IModelTileRpcInte
   public static register() { RpcManager.registerImpl(IModelTileRpcInterface, IModelTileRpcImpl); }
 
   public async requestTileTreeProps(tokenProps: IModelRpcProps, treeId: string): Promise<IModelTileTreeProps> {
-    return RequestTileTreePropsMemoizer.perform({ accessToken: RpcTrace.currentActivity!.accessToken, tokenProps, treeId });
+    if (undefined === RpcTrace.currentActivity) {
+      throw new IModelError(BentleyStatus.ERROR, "Current activity is undefined.");
+    }
+    return RequestTileTreePropsMemoizer.perform({ accessToken: RpcTrace.currentActivity.accessToken, tokenProps, treeId });
   }
 
   public async purgeTileTrees(tokenProps: IModelRpcProps, modelIds: Id64Array | undefined): Promise<void> {
     // `undefined` gets forwarded as `null`...
     if (null === modelIds)
       modelIds = undefined;
 
-    const db = await RpcBriefcaseUtility.findOpenIModel(RpcTrace.currentActivity!.accessToken, tokenProps);
+    if (undefined === RpcTrace.currentActivity) {
+      throw new IModelError(BentleyStatus.ERROR, "Current activity is undefined.");
+    }
+    const db = await RpcBriefcaseUtility.findOpenIModel(RpcTrace.currentActivity.accessToken, tokenProps);
     return db.nativeDb.purgeTileTrees(modelIds);
   }
 
   public async generateTileContent(tokenProps: IModelRpcProps, treeId: string, contentId: string, guid: string | undefined): Promise<TileContentSource> {
-    return RequestTileContentMemoizer.perform({ accessToken: RpcTrace.currentActivity!.accessToken, tokenProps, treeId, contentId, guid });
+    if (undefined === RpcTrace.currentActivity) {
+      throw new IModelError(BentleyStatus.ERROR, "Current activity is undefined.");
+    }
+    return RequestTileContentMemoizer.perform({ accessToken: RpcTrace.currentActivity.accessToken, tokenProps, treeId, contentId, guid });
   }
 
   public async retrieveTileContent(tokenProps: IModelRpcProps, key: TileContentIdentifier): Promise<Uint8Array> {
-    const db = await RpcBriefcaseUtility.findOpenIModel(RpcTrace.currentActivity!.accessToken, tokenProps);
+    if (undefined === RpcTrace.currentActivity) {
+      throw new IModelError(BentleyStatus.ERROR, "Current activity is undefined.");
+    }
+    const db = await RpcBriefcaseUtility.findOpenIModel(RpcTrace.currentActivity.accessToken, tokenProps);
     return db.tiles.getTileContent(key.treeId, key.contentId);
   }
 
   public async getTileCacheContainerUrl(_tokenProps: IModelRpcProps, id: CloudStorageContainerDescriptor): Promise<CloudStorageContainerUrl> {
     const invocation = RpcInvocation.current(this);
 
-    if (!IModelHost.usingExternalTileCache) {
+    if (undefined === IModelHost.usingExternalTileCache) {
```

Comment:
This introduces a bug. `usingExternalTileCache` is a `boolean` - your new condition will always evaluate to `false`.